Fix Exception in ArrayContains for [null]

zhenxiao opened the pull request:
Fix https://github.com/facebook/presto/issues/3328

haozhun commented on presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContains.java:
```diff
@@ -87,6 +87,10 @@ public static Boolean contains(Type type, Slice slice, Slice value)
         for (int i = 0; i < arrayBlock.getPositionCount(); i++) {
             if (arrayBlock.isNull(i)) {
                 foundNull = true;
+                if (valueBlock.isNull(0)) {
```
This if is unnecessary because this function is not nullable (see last argument on line 79). Therefore, it will never evaluate to true.

For `contains ([null], null)`, the expected return is `null`. I checked just now and this case is not covered in test.

This comments applies to other changes as well.

haozhun commented:
Add 2 tests in `TestArrayOperators`:

`assertFunction("CONTAINS(ARRAY [CAST (NULL AS BIGINT)], 1)", BOOLEAN, null);`
`assertFunction("CONTAINS(ARRAY [CAST (NULL AS BIGINT)], NULL)", BOOLEAN, null);`

zhenxiao commented:
@haozhun thanks, get comments addressed

zhenxiao commented:
@haozhun @cberner Seems the bug is also fixed by:
https://github.com/facebook/presto/commit/54a24570d440a429f9587342c91631ba69ace362
rebased the PR, only contains added testcases
Would you please take a look?

cberner commented:
Merged. Thanks!
